src/refinement/amr_criteria.cpp

```diff
@@ -55,14 +58,14 @@ AMRFirstDerivative::AMRFirstDerivative(ParameterInput *pin, std::string &block_n
 }
 
 AmrTag AMRFirstDerivative::operator()(const MeshBlockData<Real> *rc) const {
-  ParArrayND<Real> q;
   if (!rc->HasCellVariable(field) || !rc->IsAllocated(field)) {
     return AmrTag::same;
   }
 
-  q = rc->Get(field).data;
+  auto q = rc->Get(field).data.Slice(comp6, comp5, comp4, Kokkos::ALL(), Kokkos::ALL(),
+                                     Kokkos::ALL());
   std::shared_ptr<MeshBlock> pmb = rc->GetBlockPointer();
-  return Refinement::FirstDerivative(pmb.get(), q, refine_criteria, derefine_criteria);
+  return Refinement::FirstDerivative(q, refine_criteria, derefine_criteria);
```

Comment:
Clever! And also supports how, e.g., `AthenaPK` does things I assume

Reply:
Yeah, that's right.  This easily supports the state vector approach, which makes the parthenon provided criteria a lot more usable.